pkg/cloudprovider/aws/fleet/packing/packing.go

```diff
@@ -115,24 +129,26 @@ func (p *podPacker) packWithLargestPod(pods []*v1.Pod) (*Packing, []*v1.Pod) {
 	return &Packing{Pods: bestPackedPods, InstanceTypes: capacityNames}, remainingPods
 }
 
-func (p *podPacker) packPodsForCapacity(capacity *nodeCapacity, pods []*v1.Pod) (packedPods, remainingPods []*v1.Pod) {
+func (p *podPacker) packPodsForCapacity(capacity *nodeCapacity, pods []*v1.Pod) *packingsPerCapacity {
 	// start with the largest pod based on resources requested
+	resetCapacity := capacity.reserved.DeepCopy()
+	defer func() {
+		capacity.reserved = resetCapacity
```

Comment:
This seems pretty complicated and possibly error-prone (especially if there actually were concurrent access, although I realize there shouldn't be). Why not operate on a copy of `capacity` rather than modify the global copy?

Reply:
I removed this, also all this logic will be removed very soon.